sklearn/linear_model/tests/test_logistic.py

```diff
@@ -2374,3 +2374,43 @@ def test_multi_class_deprecated():
     lrCV = LogisticRegressionCV(multi_class="multinomial")
     with pytest.warns(FutureWarning, match=msg):
         lrCV.fit(X, y)
+
+
+def test_newton_cholesky_fallback_to_lbfgs(global_random_seed):
+    # Wide data matrix should lead to a rank-deficient Hessian matrix
+    # hence make the Newton-Cholesky solver raise a warning and fallback to
+    # lbfgs.
+    X, y = make_classification(
+        n_samples=10, n_features=20, random_state=global_random_seed
+    )
+    C = 1e30  # very high C to nearly disable regularization
+
+    # Check that LBFGS can converge without any warning on this problem.
+    lr_lbfgs = LogisticRegression(solver="lbfgs", C=C)
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        lr_lbfgs.fit(X, y)
+
+    n_iter_lbgs = lr_lbfgs.n_iter_[0]
+    assert n_iter_lbgs >= 1
+
+    # Check that the Newton-Cholesky solver raises a warning and falls back to
+    # LBFGS. This should also converges in the same number of iterations as the
+    # previous call since the Newton-Cholesky solver should trigger the
+    # fallback before completing the first iteration.
```

Comment:
```suggestion
    # LBFGS. This should converge with the same number of iterations as the
    # above call of lbfgs since the Newton-Cholesky triggers the fallback
    # before completing the first iteration, for the problem setting at hand.
```

Reply:
LGTM, I will accept the suggestion once the current CI run has completed to check if the way the test is written is portable enough for the different BLAS / scipy / OS kinds and versions we test against.